internal/internal_task_handlers.go

```diff
@@ -309,6 +309,11 @@ func newWorkflowTaskHandler(
 	hostEnv *hostEnvImpl,
 ) WorkflowTaskHandler {
 	ensureRequiredParams(&params)
+	// laTunnel is needed for queueResetStickinessTask
+	laTunnel := &localActivityTunnel{
+		taskCh:   make(chan *localActivityTask, 1000),
```

Comment:
What happens if someone runs 1500 local activities?

Reply:
The exceeded activities will be blocked until there is more room in the buffered channel. 
There is a go routines in Baseworker's localActivityTaskPoller continues to pull the `taskCh` and then dispatch these tasks to localActivityWorker.

`newWorkflowTaskHandler` is only called in two places.  One is in `newWorkflowWorkerInternal` https://github.com/uber-go/cadence-client/blob/4cc749b964d61fd441fa67810d43683bb2455c03/internal/internal_worker.go#L244-L261

The other one is in `replayWorkflowHistory` 
https://github.com/uber-go/cadence-client/blob/4cc749b964d61fd441fa67810d43683bb2455c03/internal/worker.go#L265

In a "`normal`" case, `newWorkflowWorkerInternal` will create a `localActivityTunnel` with a buffered channel with 1000 slots. 

https://github.com/uber-go/cadence-client/blob/4cc749b964d61fd441fa67810d43683bb2455c03/internal/internal_worker.go#L291-L294

However, I saw the comment in here

https://github.com/uber-go/cadence-client/blob/d0cf9e09a98f1a9f6fc09fab3872923f4f8c8418/internal/internal_task_handlers.go#L888-L889

I think instead of tried to initiate a laTunnel here. we should  just check if `laTunnel` is `nil`.

https://github.com/uber-go/cadence-client/blob/d0cf9e09a98f1a9f6fc09fab3872923f4f8c8418/internal/internal_task_handlers.go#L442